src/components/Favourites.js

```diff
@@ -1,18 +1,64 @@
 import SmallButton from "./SmallButton";
 import "./List.css";
 import ClinicCard from "./ClinicCard";
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { getItemsFromLocalStorage } from "./Services/ClinicStorage";
+import Searchbar from "./Searchbar";
 
 export default function Favourites() {
+  const [clinicData, setClinicData] = useState([]);
+  const [nameFilter, setNameFilter] = useState("");
+  const [showForm, setShowForm] = useState(false);
+
+  useEffect(() => {
+    const clinic = getItemsFromLocalStorage("clinicData");
+    setClinicData(clinic);
+  }, []);
+
+  function handleOnNameFilter(filterNameValue) {
+    setNameFilter(filterNameValue);
+  }
+
+  function handleToggleForm() {
+    setShowForm((prevState) => !prevState);
+  }
+
+  function renderFavourites() {
+    const savedClinic = clinicData.filter((clinicData) => {
+      return clinicData.isSaved === true;
+    });
+    const filteredSavedClinics = savedClinic.filter((clinicData) => {
+      if (nameFilter) {
+        return (
+          clinicData.name.toLowerCase().includes(nameFilter.toLowerCase()) ||
+          clinicData.place.toLowerCase().includes(nameFilter.toLowerCase()) ||
+          clinicData.notes.toLowerCase().includes(nameFilter.toLowerCase())
+        );
+      } else {
+        return true;
+      }
+    });
```

Comment:
Please consider making this function shorter and declare it somewhere else, that way the filter will be easier to read. Also, this could potentially be a reusable filter